blueprint/core/blueprint_ViewManager.cpp

```diff
@@ -52,14 +52,22 @@ namespace blueprint
     {
         const auto& [view, shadow] = getViewHandle(viewId);
 
-        view->setProperty(name, value);
-        shadow->setProperty(name, value);
-
-        // For now, we just assume that any new property update means we
-        // need to redraw or lay out our tree again. This is an easy future
-        // optimization.
-        performRootShadowTreeLayout();
-        view->repaint();
+        //TODO: There is an element of duplicate work checking ShadowView::isLayoutProperty and
+        //      then effectively running N property comparisons in the worst case again
+        //      in ShadowView::setProperty. We could have ShadowView::setProperty return a boolean
+        //      if a property was set succesfully and if false avoid performRootShadowTreeLayout ?
+        if (ShadowView::isLayoutProperty(name))
+        {
+            //TODO: Here is where we could use juce::AsyncUpdater to coalesce layout property
+            //      updates into a single root layout operation.
+            shadow->setProperty(name, value);
+            performRootShadowTreeLayout();
+        }
+        else
+        {
+            view->setProperty(name, value);
```

Comment:
Is there a need for `View` to hold the `ShadowView` properties also?

Reply:
Hmm... off the top of my head I can't think of any need. I have this feeling that it might come in handy for a `View` to know about all of its ShadowView's properties and vice versa, but the only things I can think of off hand are contrived examples like "what if I want a View that paints itself red when it's flex-direction: column," which is so obviously contrived and in reality that would just be something done at the js level in user code anyway lol

I'm cool to go ahead with this like this